src/native/minipal/time.c

```diff
@@ -74,6 +83,42 @@ int64_t minipal_hires_ticks(void)
     }
 
     return ((int64_t)(ts.tv_sec) * (int64_t)(tccSecondsToNanoSeconds)) + (int64_t)(ts.tv_nsec);
+#else
+    #error "minipal_hires_ticks requires clock_gettime_nsec_np or clock_gettime to be supported."
+#endif
+}
+
+int64_t minipal_lowres_ticks(void)
+{
+#if HAVE_CLOCK_GETTIME_NSEC_NP
+    return  (int64_t)clock_gettime_nsec_np(CLOCK_UPTIME_RAW) / (int64_t)(tccMilliSecondsToNanoSeconds);
+#elif HAVE_CLOCK_MONOTONIC
+    struct timespec ts;
+
+    // emscripten exposes CLOCK_MONOTONIC_COARSE but doesn't implement it
+#if HAVE_CLOCK_MONOTONIC_COARSE && !defined(__EMSCRIPTEN__)
+    // CLOCK_MONOTONIC_COARSE has enough precision for GetTickCount but
+    // doesn't have the same overhead as CLOCK_MONOTONIC. This allows
+    // overall higher throughput. See dotnet/coreclr#2257 for more details.
+
+    const clockid_t clockType = CLOCK_MONOTONIC_COARSE;
+#else
+    const clockid_t clockType = CLOCK_MONOTONIC;
+#endif
+
+    int result = clock_gettime(clockType, &ts);
+    if (result != 0)
+    {
+#if HAVE_CLOCK_MONOTONIC_COARSE && !defined(__EMSCRIPTEN__)
+        assert(!"clock_gettime(CLOCK_MONOTONIC_COARSE) failed");
+#else
+        assert(!"clock_gettime(CLOCK_MONOTONIC) failed");
+#endif
+    }
+
+    return ((int64_t)(ts.tv_sec) * (int64_t)(tccSecondsToMilliSeconds)) + ((int64_t)(ts.tv_nsec) / (int64_t)(tccMilliSecondsToNanoSeconds));
+#else
+    #error "minipal_lowres_ticks requires clock_gettime_nsec_np or clock_gettime to be supported."
```

Comment:
Old code had fallback to gettimeofday, I guess we assume all targeted and supported platforms that been switched over to this API supports clock_gettime_nsec_np or clock_gettime so there is no need to keep fallback?

Reply:
https://github.com/dotnet/runtime/pull/115408#discussion_r2091813199

From the [manpage](https://www.man7.org/linux/man-pages/man2/gettimeofday.2.html), `gettimeofday` is obsoleted in favor of `clock_gettime`. It shouldn't be a problem for future UNIX.